ext/io_gdml.jl

```diff
@@ -280,8 +317,71 @@ function parse_geometry(e::Ellipsoid{T,<:Any, TR, Nothing, Nothing}, x_solids::X
     nothing
 end
 
-function parse_geometry(e::World{T, 3, Cylindrical}, x_solids::XMLElement, x_define::XMLElement, id::Integer, pf::AbstractString, v::Bool)::Nothing where {T}
+@inline parse_θ(::Type{T}, θ::Nothing) where {T} = (T(0), T(360), "deg")
+@inline parse_θ(::Type{T}, θ::Tuple{T, T}) where {T} = throw(AssertionError("Torus: θ values are currently not supported"))
+function parse_geometry(e::Torus, x_solids::XMLElement, x_define::XMLElement, id::Integer, pf::AbstractString, v::Bool)::Nothing
+    if has_volume(e, x_solids, x_define, id, pf, v)
+        y = new_child(x_solids, "torus")
+        # phi, aunit = parse_φ(T, e.φ)
+        # theta1, theta2, aunit = parse_θ(T, e.θ)
+        
+        rmin, rmax = if isa(e.r_tube, Tuple)
+            e.r_tube[1], e.r_tube[2]
+        else 
+            0, e.r_tube
+        end
+
+        # rmin = 0
+
+        set_attributes(y, OrderedDict(
+            "name" => pf * string(id),
+            "rmin" => rmin,
+            "rmax" => rmax,
+            "rtor" => e.r_torus,
+            "startphi" => 0,
+            "deltaphi" => 360,
+            "lunit" => SolidStateDetectors.internal_length_unit,
+            "aunit" => "deg"
+        ))
+    
+    end
+end
+
 
+function parse_geometry(e::RegularPrism, x_solids::XMLElement, x_define::XMLElement, id::Integer, pf::AbstractString, v::Bool)::Nothing
+    if has_volume(e, x_solids, x_define, id, pf, v)
+        y = new_child(x_solids, "polyhedra")
+        
+        r = e.r
+        hZ = e.hZ
+        set_attributes(y, OrderedDict(
+            "name" => pf * string(id),
+            "startphi" => 0,
+            "deltaphi" => 360,
+            "numsides" => 6,
```

Comment:
`"numsides"` is a parametric type of `RegularPrism`:
Only `RegularPrism{T,CO,6}` is a `HexagonalPrism`.
You could extract `"numsides"` from the parametric types.

```julia
function parse_geometry(e::RegularPrism{T, <:Any, N},...) where {T, N}
    # ...
    "numsides" => N
    # ...
end
```